packages/itwinui-react/src/core/Select/Select.tsx

```diff
@@ -763,3 +774,55 @@ type MultipleSelectButtonProps<T> = {
   selectedItemsRenderer?: (options: SelectOption<T>[]) => JSX.Element;
   tagRenderer: (item: SelectOption<T>) => JSX.Element;
 };
+
+// ----------------------------------------------------------------------------
+
+const SelectListbox = React.forwardRef((props, forwardedRef) => {
+  const {
+    defaultFocusedIndex = 0,
+    autoFocus = true,
+    children: childrenProp,
+    className,
+    ...rest
+  } = props;
+
+  const [focusedIndex, setFocusedIndex] = React.useState(defaultFocusedIndex);
+
+  const autoFocusRef = React.useCallback((element: HTMLElement | null) => {
+    queueMicrotask(() => {
+      const firstFocusable = element?.querySelector(
+        '[tabindex="0"]',
+      ) as HTMLElement | null;
+      firstFocusable?.focus();
+    });
+  }, []);
+
+  const children = React.useMemo(() => {
+    return React.Children.map(childrenProp, (child, index) =>
+      React.isValidElement(child) ? (
+        <CompositeItem
+          key={index}
+          render={child}
+          ref={child.props.ref || (child as any).ref}
```

Comment:
I'm a little confused by what is the difference between the two values. Could you elaborate?

Reply:
In React 19, `ref` will be a prop. This code is basically future-proofing us while staying backwards compatible. See https://react.dev/blog/2024/04/25/react-19-upgrade-guide#deprecated-element-ref

Separately, we should create a PR updating all other instances of `element.ref` to `element.props.ref || element.ref`.